### PR TITLE
New WordPress version with persisting volume support

### PR DESCRIPTION
**Pull request by danielfree**
Based on [bitnami/wordpress](https://hub.docker.com/r/bitnami/wordpress/) image, add persisting volume support where user can choose from existing volume types.

### CODE REVIEW

File: templates/wordpress/1/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+version: '2'
+services:
+  mariadb:
+      image: 'bitnami/mariadb:latest'
```
Comment (rawmind0) [inline templates/wordpress/1/docker-compose.yml]: Please, don't use latest, instead set a concrete version.

File: templates/wordpress/1/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      volumes:
+        - 'mariadb_data:/bitnami'
+      environment:
+        - MARIADB_USER=bn_wordpress
+        - MARIADB_DATABASE=bitnami_wordpress
+        - ALLOW_EMPTY_PASSWORD=yes
+  wordpress:
+      image: 'bitnami/wordpress:latest'
```
Comment (rawmind0) [inline templates/wordpress/1/docker-compose.yml]: Please, don't use latest, instead set a concrete version.

File: templates/wordpress/1/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        - mariadb
+      environment:
+        - MARIADB_HOST=mariadb
+        - MARIADB_PORT_NUMBER=3306
+        - WORDPRESS_DATABASE_USER=bn_wordpress
+        - WORDPRESS_DATABASE_NAME=bitnami_wordpress
+        - ALLOW_EMPTY_PASSWORD=yes
+volumes:
+  mariadb_data:
+    driver: rancher-nfs
```
Comment (rawmind0) [inline templates/wordpress/1/docker-compose.yml]: Please set volume driver as variable.
```
-    driver: rancher-nfs
+    driver: ${VOLUME_DRIVER}
```

File: templates/wordpress/1/docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  wordpress_data:
+    driver: rancher-nfs
```
Comment (rawmind0) [inline templates/wordpress/1/docker-compose.yml]: Please set volume driver as variable.
```
-    driver: rancher-nfs
+    driver: ${VOLUME_DRIVER}
```

File: templates/wordpress/1/rancher-compose.yml [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+  description: "Blog tool, publishing platform and CMS based on bitnami images"
+  uuid: Wordpress-1
+  minimum_rancher_version: v0.51.0
+  questions:    
+    - variable: public_port
+      description: "public port to access the wordpress site"
+      label: "Public Port"
+      required: true
+      default: "80"
+      type: "int"
```
Comment (rawmind0) [inline templates/wordpress/1/rancher-compose.yml]: Add volume driver as question
```
+    - variable: "VOLUME_DRIVER"
+      description: "Volume driver to use with this service"
+      label: "Volume driver"
+      required: true
+      default: "local"
+      type: enum
+      options:
+        - local
+        - rancher-nfs
+        - rancher-efs
+        - rancher-ebs
```

File: templates/wordpress/1/docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+      volumes:
+        - 'wordpress_data:/bitnami'
+      depends_on:
+        - mariadb
+      environment:
+        - MARIADB_HOST=mariadb
+        - MARIADB_PORT_NUMBER=3306
+        - WORDPRESS_DATABASE_USER=bn_wordpress
+        - WORDPRESS_DATABASE_NAME=bitnami_wordpress
+        - ALLOW_EMPTY_PASSWORD=yes
```
Comment (rawmind0) [inline templates/wordpress/1/docker-compose.yml]: All the env info is hardcoded, please use questions in the rancher-compose, with these as default values.

File: templates/wordpress/1/docker-compose.yml [hunk 7]
```diff
@@ -0,0 +1,29 @@
+version: '2'
+services:
+  mariadb:
+      image: 'bitnami/mariadb:latest'
+      volumes:
+        - 'mariadb_data:/bitnami'
+      environment:
+        - MARIADB_USER=bn_wordpress
+        - MARIADB_DATABASE=bitnami_wordpress
+        - ALLOW_EMPTY_PASSWORD=yes
```
Comment (rawmind0) [inline templates/wordpress/1/docker-compose.yml]: All the info is hardcoded, please use questions in the rancher-compose, with these as default values.

File: templates/wordpress/1/docker-compose.yml [hunk 8]
```diff
@@ -0,0 +1,29 @@
+version: '2'
+services:
```
Comment (rawmind0) [inline templates/wordpress/1/docker-compose.yml]: Please, use lb instead exposing ports directly
```
+  wordpress-lb:
+    image: rancher/lb-service-haproxy:v0.6.4
+    ports:
+      - ${public_port}:${public_port}
```

File: templates/wordpress/1/docker-compose.yml [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+      volumes:
+        - 'mariadb_data:/bitnami'
+      environment:
+        - MARIADB_USER=bn_wordpress
+        - MARIADB_DATABASE=bitnami_wordpress
+        - ALLOW_EMPTY_PASSWORD=yes
+  wordpress:
+      image: 'bitnami/wordpress:latest'
+      ports:
+        - '${public_port}:80'
```
Comment (rawmind0) [inline templates/wordpress/1/docker-compose.yml]: Please, use lb instead exposing ports directly
```
-      ports:
-        - '${public_port}:80'
```

File: templates/wordpress/1/rancher-compose.yml [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+wordpress:
```
Comment (rawmind0) [inline templates/wordpress/1/rancher-compose.yml]: Please, set healthcheck and lb configuration..
```
-wordpress:
+services:
+  wordpress:-lb:
+    scale: 1
+    lb_config:
+      certs: []
+      port_rules:
+      - protocol: http
+        service: wordpress
+        source_port: ${publish_port}
+        target_port: 80
+    health_check:
+      response_timeout: 2000
+      healthy_threshold: 2
+      port: 42
+      unhealthy_threshold: 3
+  wordpress:
+    scale: 1
+    retain_ip: true
+    health_check:
+      port: 80
+      interval: 5000
+      unhealthy_threshold: 3
+      request_line: 'GET / HTTP/1.0'
+      healthy_threshold: 2
+      response_timeout: 5000
```

File: templates/wordpress/1/rancher-compose.yml [hunk 11]
```diff
@@ -0,0 +1,14 @@
+.catalog:
```
Comment (rawmind0) [inline templates/wordpress/1/rancher-compose.yml]: Please usem version 2 notation..
```
-.catalog:
+version: '2'
+catalog:
```

**Comment by rawmind0** [comment]
Hi @danielfree ...thanks for you PR...

Please, take a look to the change request..

**Comment by danielfree** [comment]
@rawmind0 thanks for your review! I'll make updates asap.

File: templates/wordpress/1/rancher-compose.yml [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+    health_check:
+      response_timeout: 2000
+      healthy_threshold: 2
+      port: 42
+      unhealthy_threshold: 3
+  wordpress:
+    scale: 1
+    retain_ip: true
+    health_check:
+     port: 80
```
Comment (rawmind0) [inline templates/wordpress/1/rancher-compose.yml]: Syntax error....
```
-     port: 80
+      port: 80
```

File: templates/wordpress/1/rancher-compose.yml [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+      type: "string"
+    - variable: mariadb_user_password
+      description: "MariaDB database user password, set on first run"
+      label: "MariaDB Database User Password"
+      required: true
+      default: "wordpress_password"
+      type: "string"
+    - variable: wordpress_username
+      description: " WordPress application username"
+      label: "WordPress application password"
```
Comment (rawmind0) [inline templates/wordpress/1/rancher-compose.yml]: Label text is wrong
```
-      label: "WordPress application password"
+      label: "WordPress application username"
```

**Comment by rawmind0** [comment]
Hi @danielfree .. Thanks for the commit...

There is one syntax error and a wrong label, please take a look at change request.

Otherwise, i'm having some errors at wordpress startup...
```
26/07/2017 18:18:28
26/07/2017 18:18:28Welcome to the Bitnami wordpress container
26/07/2017 18:18:28Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-wordpress
26/07/2017 18:18:28Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-wordpress/issues
26/07/2017 18:18:28Send us your feedback at containers@bitnami.com
26/07/2017 18:18:28
26/07/2017 18:18:32nami    INFO  Initializing apache
26/07/2017 18:18:33apache  INFO  ==> Patching httpoxy...
26/07/2017 18:18:33Error executing 'postInstallation': EEXIST: file already exists, symlink '/bitnami/apache/conf' -> '/opt/bitnami/apache/conf'
26/07/2017 18:19:02
26/07/2017 18:19:02Welcome to the Bitnami wordpress container
26/07/2017 18:19:02Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-wordpress
26/07/2017 18:19:02Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-wordpress/issues
26/07/2017 18:19:02Send us your feedback at containers@bitnami.com
26/07/2017 18:19:02
26/07/2017 18:19:06nami    INFO  Initializing apache
26/07/2017 18:19:06Error executing 'postInstallation': Module mod_version is not present in the configuration file, you need to provide a module identifier in the options parameter along with the module name
```

And some errors in maridb ...
```
26/07/2017 18:18:55Error executing 'postInstallation': ENOENT: no such file or directory, chown '/opt/bitnami/mariadb/data/test/db.opt'
26/07/2017 18:19:06
26/07/2017 18:19:06Welcome to the Bitnami mariadb container
26/07/2017 18:19:06Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-mariadb
26/07/2017 18:19:06Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-mariadb/issues
26/07/2017 18:19:06Send us your feedback at containers@bitnami.com
26/07/2017 18:19:06
26/07/2017 18:19:10nami    INFO  Initializing mariadb
26/07/2017 18:19:10mariadb INFO  ==> Cleaning data dir...
26/07/2017 18:19:10mariadb INFO  ==> Configuring permissions...
26/07/2017 18:19:30Error executing 'postInstallation': ENOENT: no such file or directory, chown '/opt/bitnami/mariadb/data/test/db.opt'
26/07/2017 18:19:41
26/07/2017 18:19:41Welcome to the Bitnami mariadb container
26/07/2017 18:19:41Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-mariadb
26/07/2017 18:19:41Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-mariadb/issues
26/07/2017 18:19:41Send us your feedback at containers@bitnami.com
26/07/2017 18:19:41
26/07/2017 18:19:45nami    INFO  Initializing mariadb
26/07/2017 18:19:45mariadb INFO  ==> Cleaning data dir...
26/07/2017 18:19:45mariadb INFO  ==> Configuring permissions...
26/07/2017 18:20:05Error executing 'postInstallation': ENOENT: no such file or directory, chown '/opt/bitnami/mariadb/data/test/db.opt'
26/07/2017 18:20:19
26/07/2017 18:20:19Welcome to the Bitnami mariadb container
26/07/2017 18:20:19Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-mariadb
26/07/2017 18:20:19Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-mariadb/issues
26/07/2017 18:20:19Send us your feedback at containers@bitnami.com
26/07/2017 18:20:19
26/07/2017 18:20:23nami    INFO  Initializing mariadb
26/07/2017 18:20:23mariadb INFO  ==> Cleaning data dir...
26/07/2017 18:20:23mariadb INFO  ==> Configuring permissions...
26/07/2017 18:20:44Error executing 'postInstallation': ENOENT: no such file or directory, chown '/opt/bitnami/mariadb/data/test/db.opt'
26/07/2017 18:20:56
26/07/2017 18:20:56Welcome to the Bitnami mariadb container
26/07/2017 18:20:56Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-mariadb
26/07/2017 18:20:56Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-mariadb/issues
26/07/2017 18:20:56Send us your feedback at containers@bitnami.com
26/07/2017 18:20:56
26/07/2017 18:21:00nami    INFO  Initializing mariadb
26/07/2017 18:21:00mariadb INFO  ==> Cleaning data dir...
26/07/2017 18:21:00mariadb INFO  ==> Configuring permissions...
26/07/2017 18:21:20Error executing 'postInstallation': ENOENT: no such file or directory, chown '/opt/bitnami/mariadb/data/test/db.opt'
26/07/2017 18:21:31
26/07/2017 18:21:31Welcome to the Bitnami mariadb container
26/07/2017 18:21:31Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-mariadb
26/07/2017 18:21:31Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-mariadb/issues
26/07/2017 18:21:31Send us your feedback at containers@bitnami.com
26/07/2017 18:21:31
26/07/2017 18:21:35nami    INFO  Initializing mariadb
26/07/2017 18:21:35mariadb INFO  ==> Cleaning data dir...
26/07/2017 18:21:35mariadb INFO  ==> Configuring permissions...
26/07/2017 18:21:55Error executing 'postInstallation': ENOENT: no such file or directory, chown '/opt/bitnami/mariadb/data/test/db.opt'
26/07/2017 18:22:06
26/07/2017 18:22:06Welcome to the Bitnami mariadb container
26/07/2017 18:22:06Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-mariadb
26/07/2017 18:22:06Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-mariadb/issues
26/07/2017 18:22:06Send us your feedback at containers@bitnami.com
26/07/2017 18:22:06
26/07/2017 18:22:10nami    INFO  Initializing mariadb
26/07/2017 18:22:10mariadb INFO  ==> Cleaning data dir...
26/07/2017 18:22:10mariadb INFO  ==> Configuring permissions...
26/07/2017 18:22:30Error executing 'postInstallation': ENOENT: no such file or directory, chown '/opt/bitnami/mariadb/data/test/db.opt'
26/07/2017 18:22:41
26/07/2017 18:22:41Welcome to the Bitnami mariadb container
26/07/2017 18:22:41Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-mariadb
26/07/2017 18:22:41Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-mariadb/issues
26/07/2017 18:22:41Send us your feedback at containers@bitnami.com
26/07/2017 18:22:41
26/07/2017 18:22:45nami    INFO  Initializing mariadb
26/07/2017 18:22:45mariadb INFO  ==> Cleaning data dir...
26/07/2017 18:22:45mariadb INFO  ==> Configuring permissions...
26/07/2017 18:23:05Error executing 'postInstallation': ENOENT: no such file or directory, chown '/opt/bitnami/mariadb/data/test/db.opt'
26/07/2017 18:23:17
26/07/2017 18:23:17Welcome to the Bitnami mariadb container
26/07/2017 18:23:17Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-mariadb
26/07/2017 18:23:17Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-mariadb/issues
26/07/2017 18:23:17Send us your feedback at containers@bitnami.com
26/07/2017 18:23:17
26/07/2017 18:23:21nami    INFO  Initializing mariadb
26/07/2017 18:23:21mariadb INFO  ==> Cleaning data dir...
26/07/2017 18:23:21mariadb INFO  ==> Configuring permissions...
26/07/2017 18:23:41Error executing 'postInstallation': ENOENT: no such file or directory, chown '/opt/bitnami/mariadb/data/test/db.opt'
26/07/2017 18:23:53
26/07/2017 18:23:53Welcome to the Bitnami mariadb container
26/07/2017 18:23:53Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-mariadb
26/07/2017 18:23:53Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-mariadb/issues
26/07/2017 18:23:53Send us your feedback at containers@bitnami.com
26/07/2017 18:23:53
26/07/2017 18:23:57nami    INFO  Initializing mariadb
26/07/2017 18:23:57mariadb INFO  ==> Cleaning data dir...
26/07/2017 18:23:57mariadb INFO  ==> Configuring permissions...
26/07/2017 18:24:17Error executing 'postInstallation': ENOENT: no such file or directory, chown '/opt/bitnami/mariadb/data/test/db.opt'
26/07/2017 18:24:29
26/07/2017 18:24:29Welcome to the Bitnami mariadb container
26/07/2017 18:24:29Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-mariadb
26/07/2017 18:24:29Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-mariadb/issues
26/07/2017 18:24:29Send us your feedback at containers@bitnami.com
26/07/2017 18:24:29
26/07/2017 18:24:33nami    INFO  Initializing mariadb
26/07/2017 18:24:33mariadb INFO  ==> Cleaning data dir...
26/07/2017 18:24:33mariadb INFO  ==> Configuring permissions...
26/07/2017 18:24:53Error executing 'postInstallation': ENOENT: no such file or directory, chown '/opt/bitnami/mariadb/data/test/db.opt'
```

Could you please take a look??

**Comment by danielfree** [comment]
I'm running the stack with my previous commit so I haven't tested the newer ones. Will check it soon. Did you clean your stack volumes before trying?

**Comment by rawmind0** [comment]
Yes, i cleaned the volumes before test...Same result in 2 different systems..

**Comment by danielfree** [comment]
@rawmind0 bugs are fixed and I've tested it with rancher-compose, could you please have another try?

**Comment by rawmind0** [comment]
Hi @danielfree .... Tried again with same result...

I got the same log errors from wordpress and mariadb services....

**Comment by danielfree** [comment]
Seems to me it's still volume issues. What type of volume are you using? I only tested rancher-nfs.

**Comment by rawmind0** [comment]
May be.... I'm using local volumes... But volumes are not shared, and just to test should work with local volumes....

**Comment by danielfree** [comment]
I'm not sure how to use local volumes, assigning two different absolute fs paths? I'll have a try tomorrow.

**Comment by rawmind0** [comment]
Just set ```volume_driver = local```. You could create absolute path (not recommended) or a name volume. Volumes are created just fine with local driver but the services gives the error that i attached.

I'll try with other volumes drivers too, just to try to figure out where the error is....

**Comment by danielfree** [comment]
hi @rawmind0 I've searched about your error and it seems to be docker/Centos/RHEL issuse.. see https://github.com/bitnami/bitnami-docker-opencart/issues/37 and https://github.com/bitnami/bitnami-docker-testlink/issues/17#issuecomment-261783035  Do you happen to run docker on Centos/RHEL with overlayfs?

**Comment by rawmind0** [comment]
Hi @danielfree ...  thanks for your advise...

I'm not running centos/rhel hosts but you are right with overlay. I've tested different docker versions with overlay and overlay2, package's not working on none of them. Also, i've tested aufs and devicemapper storage_driver, package's working fine on both. 

May be you could include a note in README.md to avoid issues about it... :)

**Comment by rawmind0** [comment]
Cool @danielfree, thanks... :)

LGTM

**Pull request merged**